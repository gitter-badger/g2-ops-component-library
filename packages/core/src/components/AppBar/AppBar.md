Cobalt Portal App Bar

```js
const SearchBar = require('../SearchBar/SearchBar')

const MOCK_ISSUE_TYPE_VALUES = [
	{
		key: 0,
		value: 'ENHANCEMENT',
		displayValue: 'enhancement',
		isSelectable: true
	},
	{
		key: 1,
		value: 'FOOBAR',
		displayValue: 'foobar',
		isSelectable: true
	},
	{
		key: 2,
		value: 'YELLOW TAXI',
		displayValue: 'yellow taxi',
		isSelectable: true
	}
]

const MOCK_PROCESS_VALUES = [
	{
		key: 0,
		value: 'ABC',
		displayValue: 'Abc',
		isSelectable: true
	},
		{
		key: 1,
		value: 'ESPN',
		displayValue: 'Espn',
		isSelectable: true
	},
]

const configCobalt = {
  type: 'cobalt',
  config: ['flag', 'role', 'yard'],
  isLoggedOn: true,
	countryCode: 'de',
	yardNumber: 5001,
  role: 'Germany Executive',
  showSearchBar: false,
  moduleName: 'Cobalt Portal'
}

const configCAS = {
  type: 'cas',
  config: ['flag', 'yard', 'phone'],
	isLoggedOn: true,
	countryCode: 'us',
  yardNumber: 12,
  phoneNumber: 7834873587,
  showSearchBar: false,
  moduleName: 'CAS Portal'
}

const logoutItems = [
  {
    key: 'userName',
    name: 'Sidharth Mehra'
  },
  {
    key: 'settings',
    name: 'Settings'
  },
  {
    key: 'logout',
    name: 'Logout'
  }
]

const searchTypes = [
  { key: 'lot', name: 'Lot' },
  { key: 'seller', name: 'Seller' },
  { key: 'owner', name: 'Owner' },
  { key: 'seller-personnel', name: 'Seller Personnel' },
  { key: 'facility', name: 'Facility' },
  { key: 'location', name: 'Location' },
  { key: 'buyer', name: 'Buyer' }
]
;<div style={{ margin: '-17px' }}>
  <AppBar
    {...configCobalt}
    onLogoutItemClicked={(event, item) => console.log(item)}
    logoutItems={logoutItems}
		onFeedbackClick={() => console.log('Feedback clicked')}
		role="foobar"
		userEmail="test@copart.com"
		selectedYard={2244}
		language="Spanglish"
		afterSendFeedback={() => { console.log('afterSendFeedback') }}
		feedbackIssueTypeValues={MOCK_ISSUE_TYPE_VALUES}
		feedbackProcessValues={MOCK_PROCESS_VALUES}
		homeYard={99}
		selectedRole="barfoo"
    renderSearchbar={() => (
      <SearchBar
        searchType={{ key: 'lot', name: 'Lot' }}
        searchTypes={searchTypes}
        borderless
        searchText="Default Search Text"
        handleSearch={() => console.log('Search Clicked')}
        showCheckbox={false}
      />
    )}
  />
</div>
```
