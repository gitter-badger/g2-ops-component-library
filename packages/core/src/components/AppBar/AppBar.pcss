
.AppBar {
	display: flex;
	align-items: center;
	padding: 0 20px !important;
	margin: 0 !important;
}

.AppBar > div {
	display: flex;
	align-items: center;
	margin: 0 !important;
}

.yardNumber {
	margin-left: 10px;
	margin-right: 7px;
}

.yardNumber > div {
	margin-left: 7px;
}

.userMenu {
	margin-left: 10px !important;
	margin-right: 10px !important;
}

.roleAndFlag {
	margin-left: 10px !important;
	margin-right: 10px !important;
}

.roleAndFlag > div {
	margin-left: 7px;
}

.feedback {
	margin: 0 !important;
}

.AppBarRight {
	margin: 0;
}