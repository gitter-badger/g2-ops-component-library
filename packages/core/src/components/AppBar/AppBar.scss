.flex-grid {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.flex-grid .col {
  display: flex;
  color: white;
  font-size: 13px;
}

.appBarLeft {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.flex-grid .col svg {
  color: white;
}

.logo {
  width: 6rem;
  margin-top: 0rem;
}

.moduleName {
  font-size: 10px;
  font-style: italic;
  color: white;
}

.text {
  margin-top: 0.6rem;
}

.iconText {
  margin-top: 0.4rem;
}

.searchBar {
  margin-top: -0.2rem;
  margin-left: 2rem;
}

.md-light {
  font-size: 30px;
  color: rgba(255, 255, 255, 1);
}

.appBarLeft > img.logo {
	display: initial;
}