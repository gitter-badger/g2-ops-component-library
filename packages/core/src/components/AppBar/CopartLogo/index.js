// @flow
export { CopartLogo } from './CopartLogo'
