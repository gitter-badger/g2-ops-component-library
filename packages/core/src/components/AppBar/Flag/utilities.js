import { COUNTRY_CODES, FLAG_URL_PREFIX, HEIGHT_22PX, HEIGHT_30PX } from './consts'

const code = (countryCode: string): string => {
    return COUNTRY_CODES[countryCode] || countryCode
}
  
export const flagUrl = (countryCode: string): string => {
    return `${FLAG_URL_PREFIX}${code(countryCode)}.png`
}

export const imageClass = (type: string) => {
    return type === 'cas'
        ? HEIGHT_22PX
        : HEIGHT_30PX;
}