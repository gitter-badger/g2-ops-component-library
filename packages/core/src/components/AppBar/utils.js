import { compose } from 'ramda'

const FIND_ALL_UNDERSCORES = /_+/g
const FIND_FIRST_LETTER_REGEX = /\w\S*/g

export const replaceUnderscoreWithSpace = (string) =>
  string && string.length > 0 && string.replace(FIND_ALL_UNDERSCORES, ' ')

export const capitalizeFirstLetter = (string) =>
  string &&
  string.replace(
    FIND_FIRST_LETTER_REGEX,
    (match) => match && `${match.charAt(0).toUpperCase()}${match.substr(1).toLowerCase()}`
  )

export const beautifyRoleText = compose(capitalizeFirstLetter, replaceUnderscoreWithSpace)