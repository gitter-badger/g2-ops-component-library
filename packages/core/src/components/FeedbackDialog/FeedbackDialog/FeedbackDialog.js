import * as React from "react";
import { Link } from "react-router-dom";

import { Button } from "components/Button";
import { TextField } from "components/TextField";
import { Checkbox } from "components/Checkbox";
import { AutoSelect } from "components/AutoSelect";
import { DialogBox } from "components/DialogBox";
import { FAQLink } from "../FAQLink";
import { FeedbackButton } from "../FeedbackButton";
import { EnhancementInfo } from "../EnhancementInfo";
import { toObject, generateFeedbackEmail, createSubjectText } from "../utilities";
import { FAQText } from '../FAQText'
import './FeedbackDialog.pcss'

import {
	FEEDBACK_ENHANCEMENT_LINK,
	ENHANCEMENT_LINK_STYLES,
	EMAIL_CHECKBOX_STYLES,
	SUBJECT_HEADER,
	OPTION_STYLES,
	PADDING_5PX_0
} from '../consts';

type PropsT = {
  userEmail: string,
  sendFeedback(): any,
  selectedYard: string | number,
  homeYard: string | number,
  selectedRole: string,
  countryCode: string,
  language: string,
  feedbackIssueTypeValues: [],
  feedbackProcessValues: string
}

const initialState = {
	open: false,
	enhancementChosen: false,
	includeEmail: true,
	feedbackValue: "",
	selectedIssueType: "",
	selectedProcess: ""
};

export class FeedbackDialog extends React.PureComponent<PropsT> {
  state = { ...initialState }

  setFeedbackValue = event => {
		event.persist();

    this.setState(state => ({
      feedbackValue: event.target.value
    }));
  };

  handleOpen = () => {
    this.setState(state => ({ open: true }));
  };

  handleClose = () => {
    this.setState(state => (initialState));
  };

  toggleIncludeEmail = event => {
    this.setState(state => ({
      includeEmail: !state.includeEmail
    }));
	};
	
	sendFeedback = async (feedback) => {
		// TODO: Handle this request here.
		// const response = await xhr.post('/feedback', feedback, {
    //   headers: { 'Content-Type': 'application/json; charset=UTF-8' },
    //   responseType: 'json',
		// })

		this.props.afterSendFeedback(feedback)
	}

  onSubmit = event => {
		event.preventDefault();
		const { props, state } = this

		this.sendFeedback({
			subject: event.target |> toObject |> createSubjectText,
			bodyContent: generateFeedbackEmail({
				email: state.includeEmail ? props.userEmail : "",
				feedback: state.feedbackValue,
				selectedYard: props.selectedYard,
				homeYard: props.homeYard,
				selectedRole: props.selectedRole,
				countryCode: props.countryCode,
				language: props.language
			})
		});

    this.handleClose();
	};
	
	setSelected = (which) => value => {
		return this.setState(state => ({ [`selected${which}`]: value }))
	}

	getDisplayOption = (whichFeedbackValues) => which => {
		return this.props[whichFeedbackValues][which.key].value;
	}

	setSelectedIssueType = option => {
		this.state.enhancementChosen = option.value === "ENHANCEMENT" ? true : false;
		this.setSelected('IssueType')(option.value)
	}

  render() {
		const { props, state } = this

    return (
      <div data-ccc="AppBarFeedback">
        <FeedbackButton onTouchTap={this.handleOpen} />
				<DialogBox
					showHeader
					showCloseButton
					title="Please provide us your feedback."
					hideDialog={!state.open}
					onDismiss={this.handleClose}
					autoScrollBodyContent
					containerClassName='copart_cc_FeedbackDialog_actionsContainer'
				>
					<FAQText />
					<form onSubmit={this.onSubmit}>
						<EmailInput email={props.userEmail} />
						<AutoSelects self={this} />
						<If condition={!state.enhancementChosen}>
							<div styleName="feedbackBox" style={PADDING_5PX_0}>
								<TextField
									placeholder="Please write your feedback."
									hintText={"Feedback with Lot Number"}
									floatingLabelText={`Feedback*`}
									onChange={this.setFeedbackValue}
									text={state.feedbackValue}
									floatingLabelFixed
									multiline
									required
									rows={4}
									fullWidth
								/>
							</div>
						</If>
						<div styleName="bottomBox">
							<If condition={!state.enhancementChosen}>
								<Checkbox
									label="Include Email"
									labelPosition="right"
									handleChange={this.toggleIncludeEmail}
									style={EMAIL_CHECKBOX_STYLES}
									defaultChecked
								/>
							</If>
							<If condition={!state.enhancementChosen}>
								<Button
									styleName="submitButton"
									label="Submit"
									type="submit"
									primary
								/>
							</If>
						</div>
						<If condition={state.enhancementChosen}>
							<EnhancementInfo />
						</If>
					</form>
				</DialogBox>
      </div>
    );
  }
}

const EmailInput = (props) => {
	return (
		<div styleName="emailInputBox">
			<TextField
				label="Email"
				styleName="emailTextField"
				floatingLabelText="Email"
				floatingLabelFixed
				type="email"
				disabled
				fullWidth
				value={props.email}
			/>
		</div>
	)
}

const AutoSelects = ({ self }) => {
	return (
		<div styleName="autoSelectsBox">
			<AutoSelect
				styleName="autoSelect"
				name="IssueType"
				options={self.props.feedbackIssueTypeValues}
				width={200}
				label="Issue Type*"
				selectedOption="2"
				required
				placeholder="Issue Type"
				value={self.state.selectedIssueType}
				onChange={self.setSelectedIssueType}
				optionStyleProps={OPTION_STYLES}
				displayOption={self.getDisplayOption('feedbackIssueTypeValues')}
			/>
			<If condition={!self.state.enhancementChosen}>
				<AutoSelect
					styleName="autoSelect"
					width={200}
					name="Process"
					label="Process*"
					placeholder="Process"
					options={self.props.feedbackProcessValues}
					selectedOption="2"
					required
					value={self.state.selectedProcess}
					onChange={self.setSelected('Process')}
					optionStyleProps={OPTION_STYLES}
					displayOption={self.getDisplayOption('feedbackProcessValues')}
				/>
			</If>
		</div>
	)
}