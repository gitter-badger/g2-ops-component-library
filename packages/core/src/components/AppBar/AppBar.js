// @flow
import type { Node, Element, ChildrenArray } from 'react'

import * as React from 'react'
import { type as getType } from 'ramda'

import MuiAppBar from 'material-ui/AppBar'
import { DefaultButton } from 'office-ui-fabric-react/lib/Button'
import { LogOutMenu } from './LogOutMenu'
import { wrapMuiContext } from 'utilities/wrapMuiContext'
import { FeedbackDialog } from 'components/FeedbackDialog'

import './AppBar.scss'
import './AppBar.pcss'

// TODO: Test
// TODO: Move into utilities file.
const allTruthy = (conditions: Function[]): Boolean => {
  // If any one of the conditions evaluates to false
  // then we want to return false, regardless of the
  // other conditionals' outcomes. Thus; the for..of
  // loop is used to exit early if we get a failing
  // condition.
  for (const condition of conditions) {
    if (!condition()) {
      return false 
    }
  }

  return true
}

type AppBarPropTypes = {
  userEmail: string,
  selectedYard: string | number,
  homeYard: string | number,
  selectedRole: string,
  language: string,
  /** IMPORTANT: Currently requires the function to actually POST the feedback
   * to the server. In the near future, AppBar/FeedbackDialog will handle this
   * functionality for you. An example of what this function should look like
   * can be found here: https://github.com/copartit/quicklooks/blob/6e0fbae22deb8f16fd90944144c4c314f7e7d441/src/components/AppBar/feedbackClick.js
   */
  afterSendFeedback(FeedbackT): any,
  /** Type of App bar, currently supports two values 'cas' and 'cobalt' */
  type: string,
  /** Config that determines elements rendered in right side of the App Bar */
  config: Array<string>,
  /** Module Name displayed under the Copart Logo */
  moduleName: string,
  /** Items to be rendered in the logout Menu */
  logoutItems?: Array<{
    key: string,
    name: string,
  }>,
  /** Two digit country code that renders the Flag */
  countryCode: string,
  /** Role Text */
  role?: string,
  /** Yard number */
  yardNumber?: number, // we remove support for string yardNumber,
  /** Phone number */
  phoneNumber?: number,
  /** To show searchbar component */
  showSearchBar?: boolean,
  /** If isLoggedOn is false, renders just the Feedback button, else everything is rendered. */
  isLoggedOn: boolean,
  /** Action to perform when a logout menu item is clicked */
  onLogoutItemClicked?: (SyntheticMouseEvent<HTMLElement>, { key: string, name: string }) => void,
  /** Action to perform on feedback button click */
  onFeedbackClick?: (SyntheticMouseEvent<HTMLElement>) => void,
  /** Render Search Bar */
  renderSearchbar?: () => Node,
  /** Override default function that renders the Flag */
  onRenderFlag: ({ countryCode: string, type: string }) => Node,
  /** Override default function that renders the Logo */
  onRenderLogo: () => Node,
  /** Takes children which will be rendered inside AppBar */
  children?: ChildrenArray<Node>,
}

const flagMapper: { [string]: string } = {
  ca: 'canada',
}

const Flag = ({ countryCode, type }) => {
  const imgProps = type === 'cas' ? { height: '22px' } : { height: '30px' }
  const flagUrl = `./public/assets/flag_icons/flag_${flagMapper[countryCode] || countryCode}.png`
  return <img src={flagUrl} alt="Flag" {...imgProps} />
}

const renderAppBarElements = (props) => {
  const {
    countryCode,
    role,
    yardNumber,
    phoneNumber,
    type,
    showSearchBar,
    logoutItems,
    onLogoutItemClicked,
    onFeedbackClick,
    onRenderFlag,
    onRenderLogo,
  } = props
  return (
    <div className="flex-grid" styleName="AppBarRight">
      <div className="col element" styleName="roleAndFlag">
        <If condition={props.config.includes('flag') && props.isLoggedOn}>
          {onRenderFlag({ countryCode, type })}
        </If>
        <If condition={props.config.includes('role') && props.isLoggedOn}>
          <div className="text role">{role}</div>
        </If>
      </div>
      <If condition={props.config.includes('yard') && props.isLoggedOn}>
        <div className="col element" styleName="yardNumber">
          <i className="material-icons">domain</i>
          <div className="iconText yardNumber">{yardNumber}</div>
        </div>
      </If>
      <div className="col" styleName="userMenu">
        <If condition={props.isLoggedOn}>
          <LogOutMenu items={logoutItems} onItemClick={onLogoutItemClicked} />
        </If>
      </div>
      <If condition={['Function'].includes(getType(props.afterSendFeedback))}>
        {/* Button + Dialog -> Won't show if afterSendFeedback is not present. */}
        <FeedbackDialog {...props} />
      </If>
    </div>
  )
}

const renderLogoAndSearchBar = ({ showSearchBar, moduleName, onRenderLogo, renderSearchbar }) => (
  <div className="flex-grid">
    <div className="appBarLeft">
      {onRenderLogo()}
      <span className="moduleName">{moduleName}</span>
    </div>
    {showSearchBar && renderSearchbar && <div className="searchBar">{renderSearchbar()}</div>}
  </div>
)

const AppBar = (props: AppBarPropTypes): Element<typeof MuiAppBar> => {
  const { config, isLoggedOn, children, ...otherProps } = props
  const {
    countryCode,
    role,
    yardNumber,
    phoneNumber,
    type,
    showSearchBar,
    logoutItems,
    onLogoutItemClicked,
    moduleName,
    renderSearchbar,
    onRenderFlag,
    onRenderLogo,
    onFeedbackClick,
    userEmail,
    selectedYard,
    language,
    afterSendFeedback,
    homeYard,
    selectedRole,
    ...appBarProps
  } = otherProps
  return (
    <MuiAppBar
      iconStyleRight={{
        marginTop: '14px',
        marginRight: '-14px',
      }}
      {...appBarProps}
      iconElementLeft={renderLogoAndSearchBar(props)}
      iconElementRight={renderAppBarElements(props)}
      data-core-component="AppBar"
    >
      {children}
    </MuiAppBar>
  )
}

AppBar.defaultProps = {
  onRenderFlag: ({ countryCode, type }: { countryCode: string, type: string }): Node => (
    <Flag countryCode={countryCode} type={type} />
  ),
  onRenderLogo: (): Node => <img className="logo" src="./public/assets/images/logo.svg" alt="Copart" />,
  config: ['flag', 'role', 'yard']
}

const wrappedAppBar = wrapMuiContext(AppBar)
export { wrappedAppBar as AppBar }