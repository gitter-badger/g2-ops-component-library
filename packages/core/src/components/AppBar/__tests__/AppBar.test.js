import React from 'react'
import { mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { FeedbackDialog } from 'components/FeedbackDialog'
import { wrapWithMaterialUIContext } from 'utilities/wrapWithContext'

import { AppBar } from 'components/AppBar'

const getProps = (extraProps = {}) => ({
  config: [],
  type: 'CAS',
  moduleName: 'CAS Portal',
  countryCode: 'us',
  isLoggedOn: false,
  ...extraProps,
})

describe('<AppBar />', () => {
  test('<AppBar /> renders properly', () => {
    expect(renderer.create(wrapWithMaterialUIContext(<AppBar {...getProps()} />))).toMatchSnapshot()
	})
	
	test('should not render <FeedbackDialog /> if specific props are invalid', () => {
		const wrapper = mount(<AppBar {...getProps()} />)
		expect(wrapper.find(FeedbackDialog).length).toEqual(0)
	})

	test('should render <FeedbackDialog /> if specific props are valid', () => {
		const MOCK_ISSUE_TYPE_VALUES = [
			{
				key: 0,
				value: 'ENHANCEMENT',
				displayValue: 'enhancement',
				isSelectable: true
			},
			{
				key: 1,
				value: 'FOOBAR',
				displayValue: 'foobar',
				isSelectable: true
			},
			{
				key: 2,
				value: 'YELLOW TAXI',
				displayValue: 'yellow taxi',
				isSelectable: true
			}
		]

		const MOCK_PROCESS_VALUES = [
			{
				key: 0,
				value: 'ABC',
				displayValue: 'Abc',
				isSelectable: true
			},
				{
				key: 1,
				value: 'ESPN',
				displayValue: 'Espn',
				isSelectable: true
			},
		]

		const wrapper = mount(<AppBar
			config={[]}
			type={'CAS'}
			moduleName={'CAS Portal'}
			countryCode={'us'}
			isLoggedOn={false}
			userEmail={'STUB'}
			afterSendFeedback={jest.fn()}
			selectedYard={60}
			homeYard={'STUB'}
			selectedRole={'STUB'}
			countryCode={'ST'}
			language={'STUB'}
			feedbackIssueTypeValues={MOCK_ISSUE_TYPE_VALUES}
			feedbackProcessValues={MOCK_PROCESS_VALUES}
		/>)

		expect(wrapper.find('FeedbackDialog').length).toEqual(1)
	})

  test('should render LogOut Menu when isLoggedOn=true', () => {
		const tree = mount(wrapWithMaterialUIContext(<AppBar {...getProps({ isLoggedOn: true })} />))
    expect(tree.find('LogOutMenu').exists()).toBe(true)
  })

  test('should render different components based on config', () => {
    const configPropToComponentMap = {
      role: 'div.role',
      yard: 'div.yardNumber',
      flag: 'Flag',
    }
    Object.keys(configPropToComponentMap).forEach((config) => {
			const tree = mount(wrapWithMaterialUIContext(<AppBar {...getProps({ config: [ config ], isLoggedOn: true })} />))
      expect(tree.find(configPropToComponentMap[config]).exists()).toBe(true)
    })
  })

  describe('<SearchBar />', () => {
    test('should render search bar when both showSearchBar=true & renderSearchbar is defined ', () => {
      const renderSearchbar = jest.fn(() => () => <span className="searchBar" />)
      const configsToRender = [
        {
          props: { showSearchBar: true, renderSearchbar },
          exists: true,
        },
        {
          props: { showSearchBar: false, renderSearchbar },
          exists: false,
        },
        {
          props: { showSearchBar: true },
          exists: false,
        },
      ]
      configsToRender.forEach((config) => {
        const tree = mount(wrapWithMaterialUIContext(<AppBar {...getProps(config.props)} />))
        expect(tree.find('div.searchBar').exists()).toEqual(config.exists)
      })
    })
  })
})
