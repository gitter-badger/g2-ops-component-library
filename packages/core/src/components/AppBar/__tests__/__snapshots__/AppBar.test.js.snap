// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> <AppBar /> renders properly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#1d5ab9",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <div
      className="flex-grid"
    >
      <div
        className="appBarLeft"
      >
        <img
          alt="Copart"
          className="logo"
          src="./public/assets/images/logo.svg"
        />
        <span
          className="moduleName"
        >
          CAS Portal
        </span>
      </div>
    </div>
  </div>
  <h1
    onClick={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    
  </h1>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "-14px",
        "marginTop": "14px",
        "muiPrepared": true,
      }
    }
  >
    <div
      className="flex-grid"
    >
      <div
        className="col element"
      />
      <div
        className="col"
      />
      <button
        aria-describedby={null}
        aria-label={undefined}
        aria-labelledby={null}
        aria-pressed={undefined}
        className="ms-Button ms-Button--default root-37"
        data-is-focusable={true}
        disabled={undefined}
        onClick={undefined}
        type="button"
      >
        <div
          className="ms-Button-flexContainer flexContainer-38"
        >
          <div
            className="ms-Button-textContainer textContainer-39"
          >
            <div
              className="ms-Button-label label-41"
              id="id__0"
            >
              Feedback
            </div>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
